### PR TITLE
Split lesson parts

### PR DESCRIPTION
**Pull request by kicferk1**
### Context
Lessons are long. We want to split them.
Here is the UI to split them.

### Changes proposed in this pull request
1. Add a button and a page for splitting a lesson part.
2. Add endpoints for splitting a lesson part.
3. Add controller methods and policies for splitting a lesson part.

### Guidance to review
I haven't added lesson part navigation yet. That's next PR.

I opted to use more space than govuk pages use on the split page. That's because I think it is really convenient to see the old and new lesson part side-by-side. And for that, 960px that gov pages believe in, is too narrow. I intend to show it to Mark A (who will likely be the person who interacts with it the most) and see if he likes it.

image

### Testing
1. Not testing the presence of the button.
2. Tested with request tests (rspec).
3. Tested with request tests and unit tests (rspec).

### CODE REVIEW

File: app/views/core_induction_programme/lesson_parts/show_split.html.erb [hunk 1]
```diff
@@ -0,0 +1,57 @@
+<% content_for :outside_container do %>
+<div class="govuk-width-container govuk-main-wrapper govuk-body">
+  <%= govuk_breadcrumbs breadcrumbs: course_lesson_breadcrumbs(@current_user, @course_lesson_part.course_lesson) %>
+  <h1 class="govuk-heading-l">Split lesson part</h1>
+  <ul>
+    <li>
+      The part on the left hand side is the one you are splitting (current).
```
Comment (callumacrae) [inline app/views/core_induction_programme/lesson_parts/show_split.html.erb:7]: this is very very inaccessible

Comment (kicferk1) [reply app/views/core_induction_programme/lesson_parts/show_split.html.erb:7]: Eh, it's meant to be internal. 
You mean the list is inaccessible? Or left/right distinction?

Comment (callumacrae) [reply app/views/core_induction_programme/lesson_parts/show_split.html.erb:7]: left/right distinction! both the words and the actual layout

Comment (kicferk1) [reply app/views/core_induction_programme/lesson_parts/show_split.html.erb:7]: Well, truth be told I think there is going to be one unlucky user of that page, which is Mark A. 

I don't think there is a very accessible way of handling this problem to begin with. The problem being 'splitting very long markdown roughly in half, in a text area, and showing what the outcome will look like'.

The amount of content to show is massive, so much so that showing it in a column, one after another, looked really unintuitive. That's why I opted to use visuals to show them in parallel.

I can rephrase it to be "The first part is the one you are splitting, the second part is new" or something like that if you think it's much better.

Comment (callumacrae) [reply app/views/core_induction_programme/lesson_parts/show_split.html.erb:7]: if it's for internal usage only I'm not sure it matters that much :)

**Review by amazimbe — APPROVED** [review]

**Review by rupollbabu — APPROVED** [review]

File: app/views/core_induction_programme/lesson_parts/show_split.html.erb (hunk 1 above)
Comment (kicferk1) [reply app/views/core_induction_programme/lesson_parts/show_split.html.erb:7]: Alright :)

**Review by callumacrae — APPROVED** [review]
